packages/runtime/src/widgets/DataGrid/DataGrid.tsx

```diff
@@ -272,12 +277,14 @@ export const DataGrid: React.FC<GridProps> = (props) => {
               }
               render={(_: unknown, record: unknown): ReactElement => {
                 return (
-                  <DataCell
-                    columnIndex={index}
-                    data={record}
-                    scopeName={itemTemplate.name}
-                    template={itemTemplate.template}
-                  />
+                  <CustomScopeProvider value={record as CustomScope}>
```

Comment:
You don't need this extra provider. The record is passed in as a prop to the data cell

Reply:
I think we need this one, because when we use the itemTemplate inside the DataRow, then in the DataCell, we are getting expression value and then we need to use `useTemplateData` here to convert this expression into the data. 

If I am not using this extra provider, then I am getting the error while `useTemplateData` that the variable is undefined.

Reply:
Instead of injecting an extra scope provider, let's expose an extra `context` option on useTemplateData and pass in the record that way. This will avoid having a bunch of extra providers.

Reply:
So basically when we pass the optional context to useTemplateData, then we need to merge that optional context with the useTemplateData's custom scope ?

Reply:
we merge that context inside the evaluate hook, yes.